tla-pp/src/test/scala/at/forsyte/apalache/tla/pp/TestTemporalEncoder.scala

```diff
@@ -0,0 +1,180 @@
+package at.forsyte.apalache.tla.pp
+
+import at.forsyte.apalache.tla.lir.BoolT1
+import at.forsyte.apalache.tla.lir.OperEx
+import at.forsyte.apalache.tla.lir.TlaEx
+import at.forsyte.apalache.tla.lir.TlaLevelFinder
+import at.forsyte.apalache.tla.lir.TlaLevelTemporal
+import at.forsyte.apalache.tla.lir.TlaModule
+import at.forsyte.apalache.tla.lir.TlaOperDecl
+import at.forsyte.apalache.tla.lir.Typed
+import at.forsyte.apalache.tla.lir.ValEx
+import at.forsyte.apalache.tla.lir.oper.TlaBoolOper
+import at.forsyte.apalache.tla.lir.oper.TlaOper
+import at.forsyte.apalache.tla.lir.oper.TlaTempOper
+import at.forsyte.apalache.tla.lir.transformations.impl.IdleTracker
+import at.forsyte.apalache.tla.lir.values.TlaBool
+import at.forsyte.apalache.tla.pp.temporal.LoopEncoder
+import at.forsyte.apalache.tla.pp.temporal.TableauEncoder
+import at.forsyte.apalache.tla.typecomp.ScopedBuilder
+import org.junit.runner.RunWith
+import org.scalacheck.Gen
+import org.scalacheck.Gen.oneOf
+import org.scalacheck.Prop
+import org.scalacheck.Prop.AnyOperators
+import org.scalacheck.Prop.forAll
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatestplus.junit.JUnitRunner
+import org.scalatestplus.scalacheck.Checkers
+
+@RunWith(classOf[JUnitRunner])
+class TestTemporalEncoder extends AnyFunSuite with Checkers {
+  private val loopEncoder = new LoopEncoder(new IdleTracker())
+
+  // loop encoder expects init and next declarations, so we generate empty ones to use when we don't care about them
+  val init = new TlaOperDecl("init", List.empty, new ValEx(new TlaBool(true))(Typed(BoolT1)))(Typed(BoolT1))
+  val next = new TlaOperDecl("next", List.empty, new ValEx(new TlaBool(true))(Typed(BoolT1)))(Typed(BoolT1))
+  val module = new TlaModule("module", List(init, next))
+  val modWithPreds = loopEncoder.addLoopLogic(module, init, next)
+
+  val levelFinder = new TlaLevelFinder(modWithPreds.module)
+
+  val builder = new ScopedBuilder()
+
+  private val tableauEncoder =
+    new TableauEncoder(
+        modWithPreds.module,
+        new UniqueNameGenerator(),
+        loopEncoder,
+        new IdleTracker(),
+    )
+
+  // ad hoc builder for well-typed temporal expressions (everything is typed boolean)
```

Comment:
once there exists a typed generator, this ad-hoc implementation could be replaced, to enable more extensive tests